phpBB/phpbb/language/language_file_loader.php

```diff
@@ -0,0 +1,226 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\language;
+
+use \phpbb\language\exception\language_file_not_exists;
+
+/**
+ * Language file loader
+ */
+class language_file_loader
+{
+	/**
+	 * @var string	Path to phpBB's root
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * @var string	Extension of PHP files
+	 */
+	protected $php_ext;
+
+	/**
+	 * @var \phpbb\extension\manager	Extension manager
+	 */
+	protected $extension_manager;
+
+	/**
+	 * Constructor
+	 *
+	 * @param string	$phpbb_root_path	Path to phpBB's root
+	 * @param string	$php_ext			Extension of PHP files
+	 */
+	public function __construct($phpbb_root_path, $php_ext)
+	{
+		$this->phpbb_root_path	= $phpbb_root_path;
+		$this->php_ext			= $php_ext;
+
+		$this->extension_manager = null;
+	}
+
+	/**
+	 * Extension manager setter
+	 *
+	 * @param \phpbb\extension\manager	$extension_manager	Extension manager
+	 *
+	 * @return null
+	 */
+	public function set_extension_manager(\phpbb\extension\manager $extension_manager)
+	{
+		$this->extension_manager = $extension_manager;
+	}
+
+	/**
+	 * Loads language array for the given component
+	 *
+	 * @param string		$component	Name of the language component
+	 * @param string|array	$locale		ISO code of the language to load, or array of ISO codes if you want to
+	 * 									specify additional language fallback steps
+	 * @param array			$lang		Array reference containing language strings
+	 *
+	 * @return null
+	 */
+	public function load($component, $locale, &$lang)
+	{
+		if (!is_array($locale))
+		{
+			$locale = array($locale);
+		}
+
+		// Determine path to language directory
+		$path = $this->phpbb_root_path . 'language/';
+
+		$this->load_file($path, $component, $locale, $lang);
+	}
+
+	/**
+	 * Loads language array for the given extension component
+	 *
+	 * @param string		$extension	Name of the extension
+	 * @param string		$component	Name of the language component
+	 * @param string|array	$locale		ISO code of the language to load, or array of ISO codes if you want to
+	 * 									specify additional language fallback steps
+	 * @param array			$lang		Array reference containing language strings
+	 *
+	 * @return null
```

Comment:
fixed (github doesn't show it as an outdated diff for some reason, at least not at my end)